updatecli/updatecli.d/letsencrypt_cert.ci.yaml

```diff
@@ -0,0 +1,61 @@
+name: "Bump lets_encrypt_dns_challenged_domains `expiry` date for cert.ci"
+
+scms:
+  default:
+    kind: github
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.repository }}"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      branch: "{{ .github.branch }}"
+
+sources:
+  currentExpiryCert:
+    name: Get current lets_encrypt_dns_challenged_domains `expiry` date for cert.ci
+    kind: terraform/file
+    spec:
+      file: locals.tf
+      path: 'locals.lets_encrypt_dns_challenged_domains'
+    transformers:
+      - findsubmatch:
+          pattern: '"cert.ci.jenkins.io".*=.(.*)'
+          captureindex: 1
+  nextExpiry:
+    name: Prepare next lets_encrypt_dns_challenged_domains `expiry` date within 3 months
+    kind: shell
+    spec:
+      command: bash ./updatecli/scripts/dateadd.sh
+      environments:
+        - name: PATH
+
+conditions:
+  checkIfExpirySoonExpiredcert:
+    kind: shell
+    sourceid: currentExpiryCert
+    spec:
+      command: bash ./updatecli/scripts/datediff.sh # current expiry date value passed as argument
+      environments:
+        - name: PATH
+
+targets:
+  updateNextExpirycert:
+    name: Update Terraform file `locals.tf` with new expiration date for cert.ci
+    kind: file
```

Comment:
as per https://github.com/updatecli/updatecli/issues/1859
need to use matchpattern and replacepattern for now

Reply:
Could you add a code comment for it? A PR comment is hard to track when looking at the code (for futures selves) ;)